Source/ProgressionSystemRuntime/Private/Data/PSWorldSubsystem.cpp

```diff
@@ -0,0 +1,161 @@
+// Copyright (c) Valerii Rotermel & Yevhenii Selivanov
+
+#include "Data/PSWorldSubsystem.h"
+
+#include "Controllers/MyPlayerController.h"
+#include "Data/PSDataAsset.h"
+#include "Data/PSSaveGameData.h"
+#include "Kismet/GameplayStatics.h"
+#include "LevelActors/PlayerCharacter.h"
+#include "MyDataTable/MyDataTable.h"
+#include "MyUtilsLibraries/UtilsLibrary.h"
+#include "UtilityLibraries/MyBlueprintFunctionLibrary.h"
+
+#include UE_INLINE_GENERATED_CPP_BY_NAME(PSWorldSubsystem)
+
+// Returns this Subsystem, is checked and wil crash if can't be obtained
+UPSWorldSubsystem& UPSWorldSubsystem::Get()
+{
+	const UWorld* World = UUtilsLibrary::GetPlayWorld();
+	checkf(World, TEXT("%s: 'World' is null"), *FString(__FUNCTION__));
+	UPSWorldSubsystem* ThisSubsystem = World->GetSubsystem<ThisClass>();
+	checkf(ThisSubsystem, TEXT("%s: 'ProgressiorSubsystem' is null"), *FString(__FUNCTION__));
+	return *ThisSubsystem;
+}
+
+// Returns this Subsystem, is checked and wil crash if can't be obtained
+UPSWorldSubsystem& UPSWorldSubsystem::Get(const UObject& WorldContextObject)
+{
+	const UWorld* World = GEngine->GetWorldFromContextObjectChecked(&WorldContextObject);
+	checkf(World, TEXT("%s: 'World' is null"), *FString(__FUNCTION__));
+	UPSWorldSubsystem* ThisSubsystem = World->GetSubsystem<ThisClass>();
+	checkf(ThisSubsystem, TEXT("%s: 'ProgressiorSubsystem' is null"), *FString(__FUNCTION__));
+	return *ThisSubsystem;
+}
+
+const FPSRowData& UPSWorldSubsystem::GetCurrentRow() const
+{
+	const UPSSaveGameData* SaveGameInstance = GetCurrentSaveGameData();
+	checkf(SaveGameInstance, TEXT("ERROR: 'SaveGameInstanceInternal' is null"));
+
+	return SaveGameInstance->GetCurrentRow();
+}
+
+// Returns the data asset that contains all the assets of Progression System game feature
+const UPSDataAsset* UPSWorldSubsystem::GetPSDataAsset() const
+{
+	return PSDataAssetInternal.LoadSynchronous();
+}
+
+void UPSWorldSubsystem::SetHUDComponent(UPSHUDComponent* MyHUDComponent)
+{
+	checkf(MyHUDComponent, TEXT("%s: My progression system component is null"), *FString(__FUNCTION__));
+	PSHUDComponentInternal = MyHUDComponent;
+}
+
+void UPSWorldSubsystem::OnWorldBeginPlay(UWorld& InWorld)
+{
+	Super::OnWorldBeginPlay(InWorld);
+
+	// Listen events on player type changed and Character spawned
+	if (APlayerCharacter* MyPlayerCharacter = UMyBlueprintFunctionLibrary::GetLocalPlayerCharacter())
+	{
+		MyPlayerCharacter->OnPlayerTypeChanged.AddUniqueDynamic(this, &ThisClass::OnPlayerTypeChanged);
+	}
+	else if (AMyPlayerController* MyPC = UMyBlueprintFunctionLibrary::GetLocalPlayerController())
+	{
+		MyPC->GetOnNewPawnNotifier().AddUObject(this, &ThisClass::OnCharacterPossessed);
+	}
+
+	// requires to be executed to guarntee that data asset will be loaded from the ini file
+	// otherwise the data asset file will not be loaded at fresh (heaviest) load of UE
+	PSDataAssetInternal = GetPSDataAsset();
```

Comment:
I see the comment above, but original issue you use cached `PSDataAssetInternal` property.
Follow next tasks and you will not experience this issue anymore:
1. Replace all usage of `PSDataAssetInternal->XXX` property by `UPSDataAsset::Get().XXX`(see another comment)
2. Completely remove cached `PSDataAssetInernal` property.

As a result, all usages of Data Asset will always follow the loaded file, even for fresh (heaviest) load of UE.

I appreciate if you start commenting all such workarounds as `// #HACK comment`, so I will not miss them and help to define the issue.

Reply:
> Replace all usage of PSDataAssetInternal->XXX property by UPSDataAsset::Get().XXX. There is already comment with this task: https://github.com/h4rdmol/ProgressionSystem/pull/1#discussion_r1493577275.
Done. 

> Completely remove cached PSDataAssetInernal property.
I can not remove it, since it configuration instead of cache. 
![image](https://github.com/h4rdmol/ProgressionSystem/assets/5227233/a72bf8f8-308b-45c0-b2a4-c12506ed5331)

> I appreciate if you start commenting all such workarounds as // #HACK comment, so I will not miss them and help to define the issue.
Since the hack is required, i've improved the comment

Reply:
Since you addressed original issue, there is no sense in this hack anymore. Please remove it: you can call `LoadGameFromSave()` without the hack.Why?

Reply:
hack removed.